lighthouse-core/lib/stack-packs.js

```diff
@@ -0,0 +1,52 @@
+/**
+ * @license Copyright 2019 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+const stackPacks = require('@lighthouse/stack-packs');
+
+const stackPacksToInclude = [{
+  packId: 'wordpress',
+  npmOrIconName: 'wordpress',
+  requiredStacks: ['js:wordpress']
+}];
```

Comment:
should this info be in `@lighthouse/stack-packs'` and this file can just look it up by `packId`?

Reply:
also previous discussion. 

* we have an id provided in the stackpack module
* we have an `id` of some sort coming from artifacts.Stacks. 
* the stackpack item (via `stackPacksToInclude`) has 1+ things required to discover in order to show it.

i like the idea that the stackpacks module/repo doesnt know anything about `requiredStacks`. and we define it here when we say "okay we're accepting this stackpack." how does that aspect sound, @brendankenny ?

Reply:
> i like the idea that the stackpacks module/repo doesnt know anything about requiredStacks. and we define it here when we say "okay we're accepting this stackpack." how does that aspect sound, @brendankenny?

ok, separation like that makes sense. Maybe we could still clear up all the various IDs running around?

for `stackPacksToInclude` we have `requiredStacks`, the list of things that must be in `artifacts.Stacks` to trigger inclusion. If they're all found in the artifact, the stack pack is found by matching `npmOrIconName` against the `id` found in each pack in `@lighthouse/stack-packs`.

Can we delete (the current) `packId` -- I don't see how it'll ever be anything except the id of the stack pack to use, and since those have to be unique it seems fine to use that here to identify both

And maybe I take back what I said before and `detectedId` isn't the best name for `npmOrIconName`, since it's not what's detected, it's the id of the stack pack to use if `requiredStacks` are all detected. We could use the name `packId` for `npmOrIconName` instead?